core/interface.go

```diff
@@ -126,3 +128,50 @@ type Logger interface {
 	LogIfError(err error, args ...interface{})
 	IsInterfaceNil() bool
 }
+
+// MessageP2P defines what a p2p message can do (should return)
+type MessageP2P interface {
+	From() []byte
+	Data() []byte
+	Payload() []byte
+	SeqNo() []byte
+	Topic() string
+	Signature() []byte
+	Key() []byte
+	Peer() PeerID
+	Timestamp() int64
+	IsInterfaceNil() bool
+}
+
+// InterceptedDebugger defines an interface for debugging the intercepted data
```

Comment:
mixed feelings about declaring this interface here. Also, for the Iterceptor and EpochStartActionHandler as they do not quite define the core business logic of elrond-go.

Reply:
will remain as it is, for now